src/Compilers/CSharp/Portable/Symbols/Source/SourceCustomEventSymbol.cs

```diff
@@ -72,51 +72,92 @@ internal sealed class SourceCustomEventSymbol : SourceEventSymbol
 
             AccessorDeclarationSyntax addSyntax = null;
             AccessorDeclarationSyntax removeSyntax = null;
-            foreach (AccessorDeclarationSyntax accessor in syntax.AccessorList.Accessors)
+
+            if (syntax.AccessorList != null)
             {
-                switch (accessor.Kind())
+                foreach (AccessorDeclarationSyntax accessor in syntax.AccessorList.Accessors)
+                {
+                    bool checkBody = false;
+
+                    switch (accessor.Kind())
+                    {
+                        case SyntaxKind.AddAccessorDeclaration:
+                            if (addSyntax == null)
+                            {
+                                addSyntax = accessor;
+                                checkBody = true;
+                            }
+                            else
+                            {
+                                diagnostics.Add(ErrorCode.ERR_DuplicateAccessor, accessor.Keyword.GetLocation());
+                            }
+                            break;
+                        case SyntaxKind.RemoveAccessorDeclaration:
+                            if (removeSyntax == null)
+                            {
+                                removeSyntax = accessor;
+                                checkBody = true;
+                            }
+                            else
+                            {
+                                diagnostics.Add(ErrorCode.ERR_DuplicateAccessor, accessor.Keyword.GetLocation());
+                            }
+                            break;
+                        case SyntaxKind.GetAccessorDeclaration:
+                        case SyntaxKind.SetAccessorDeclaration:
+                            diagnostics.Add(ErrorCode.ERR_AddOrRemoveExpected, accessor.Keyword.GetLocation());
+                            break;
+
+                        case SyntaxKind.UnknownAccessorDeclaration:
+                            // Don't need to handle UnknownAccessorDeclaration.  An error will have 
+                            // already been produced for it in the parser.
+                            break;
+
+                        default:
+                            throw ExceptionUtilities.UnexpectedValue(accessor.Kind());
+                    }
+
+                    if (checkBody && !IsAbstract && accessor.Body == null && accessor.ExpressionBody == null && accessor.SemicolonToken.Kind() == SyntaxKind.SemicolonToken)
```

Comment:
>accessor.SemicolonToken.Kind() == SyntaxKind.SemicolonToken [](start = 112, length = 59)

This seems to depend on how the parser handles error recovery in a fragile way.  I suggest `accessor.Body == null` instead.

Reply:
> This seems to depend on how the parser handles error recovery in a fragile way.  Why isn't the previous accessor.Body == null sufficient?

This duplicates exact conditions under which the parser used to reported this error before. I do not intend to report it under any new conditions. Different conditions would cause different errors reported in the parser. For example, if semicolon is missing, the parser would complain about missing '''{```.

---
In reply to: [285299111](https://github.com/dotnet/roslyn/pull/35756#discussion_r285299111) [](ancestors = 285299111)